src/components/DataSearch.jsx

```diff
@@ -0,0 +1,365 @@
+import React, { Component } from 'react';
+import {
+  bool,
+  string,
+  object,
+  number,
+  func,
+  dataField,
+  position,
+  suggestions,
+  highlightField,
+  queryFormat,
+  fuzziness,
+  title,
+  any,
+  react,
+  themePreset
+} from '../utils/types';
+import Input, { suggestionsContainer, suggestions as suggestionsCss } from './Input';
+import Container from '../styles/Container';
+import Title from '../styles/Title';
+import { getClassName, getComponent, hasCustomRenderer } from '../utils/helper';
+import Downshift from 'downshift';
+import Icons from './Icons';
+import Loader from './Loader';
+import Error from './Error';
+import SuggestionItem from '../addons/SuggestionItem';
+import NoSuggestions from './NoSuggestions';
+import Searchbase from '@appbaseio/searchbase';
+
+class DataSearch extends Component {
+  constructor(props) {
+    super(props);
+    const { index, url, dataField, credentials } = props;
+    const currentValue = props.value || props.defaultValue || '';
+
+    this.state = {
+      currentValue,
+      suggestions: [],
+      isOpen: false
+    };
+
+    this.searchBase = new Searchbase({
+      index,
+      url,
+      dataField,
+      credentials
+    });
+
+    this.searchBase.subscribeToStateChanges(() => {
```

Comment:
Why we are doing `forceUpdate` ?

Can we not do it with setting the state variables instead of relying on context ( `this` ) variable?

Reply:
true